### PR TITLE
chore: update README to note requirement for self-hosted runners using v4

### PR DESCRIPTION
**Pull request by tim-finnigan**
*Issue #, if available:* https://github.com/aws-actions/configure-aws-credentials/issues/920#issuecomment-1832694323

*Description of changes:* Update README to note requirement for self-hosted runners using v4

---

* [ x ] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws-actions/configure-aws-credentials/blob/main/CONTRIBUTING.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ using an IAM role to generate temporary security credentials.
 #### v4 Announcement (9/11/23)
 
 We have just released `v4` of Configure AWS Credentials. The only thing that
-changed from `v3` is that the action now runs on `node20` instead of `node16`.
+changed from `v3` is that the action now runs on `node20` instead of `node16`. If using a self-hosted runner, specify [v2.311.0](https://github.com/actions/runner/releases/tag/v2.311.0) or above for `node20` support.
```
Comment (kellertk) [inline README.md:23]: No one will notice this when it's rendered (so don't change it) but the line wrapping doesn't match the other line wrappings.

**Review by kellertk — CHANGES_REQUESTED** [review]
I don't understand why the licenses got updated if you're only making a README change. Did you add THIRD-PARTY to your commit by mistake?

**Comment by tim-finnigan** [comment]
> I don't understand why the licenses got updated if you're only making a README change. Did you add THIRD-PARTY to your commit by mistake?

That file just updated after running `npm run package`. Should I undo any changes to that file?

**Comment by tim-finnigan** [comment]
> I don't understand why the licenses got updated if you're only making a README change. Did you add THIRD-PARTY to your commit by mistake?

Reverted the license changes happened after packaging.

**Review by kellertk — APPROVED** [review]

**Pull request merged**